shim.cpp

```diff
@@ -50,12 +51,20 @@ namespace std
     typedef optional<wstring> wstring_p;
 }
 
+std::wstring GetDirectory(wchar_t* exe) {
```

Comment:
I would use `wstring_view` to minimize the creation of new strings as much as possible
```suggestion
std::wstring_view GetDirectory(std::wstring_view exe) {
```